### PR TITLE
Add user accommodation filter

### PR DESCRIPTION
**Pull request by Bozydarek**
Add simple accommodation filter with for options:

1. No accommodation
2. Not continuous accommodation
3. Partial accommodation
4. Full accommodation

### CODE REVIEW

File: app/src/users/admin.py [hunk 1]
```diff
@@ -116,6 +116,44 @@ def queryset(self, request, queryset):
...
+            ('not_continuous', "Not continuous accommodation"),
+            ('partial', "Partial accommodation"),
+            ('full', "Full accommodation"),
+        ]
+
+    def queryset(self, request, queryset):
+        value = self.value()
+        if value == 'no':
+            return queryset.filter(
+                Q(preferences__accommodation_day_1=False) &
```
Comment (ref-humbold) [inline app/src/users/admin.py:135]: I think you don't have to use `Q` objects with `&`, keyword arguments have the same effect, see https://docs.djangoproject.com/en/3.2/ref/models/querysets/#filter

File: app/src/users/admin.py [hunk 2]
```diff
@@ -116,6 +116,44 @@ def queryset(self, request, queryset):
...
+                Q(preferences__accommodation_day_2=False) &
+                Q(preferences__accommodation_day_3=False))
+        if value == 'not_continuous':
+            return queryset.filter(
+                Q(preferences__accommodation_day_1=True) &
+                Q(preferences__accommodation_day_2=False) &
+                Q(preferences__accommodation_day_3=True))
+        if value == 'partial':
+            return queryset.filter(
+                Q(preferences__accommodation_day_1=True) |
```
Comment (ref-humbold) [inline app/src/users/admin.py:145]: What was you exact intention for `partial`? This filter suggests, that it means "people that booked at least one night during ZOSIA".

Comment (ref-humbold) [inline app/src/users/admin.py:145]: What was you exact intention for `partial`? This filter suggests, that it means "people that booked at least one night during ZOSIA".

**Review by ref-humbold — CHANGES_REQUESTED** [review]

File: app/src/users/admin.py (hunk 2 above)
Comment (Bozydarek) [reply app/src/users/admin.py:145]: Yes, the filter is correct. If you have some idea one how to name this better, I’m open to suggestions. :)

File: app/src/users/admin.py (hunk 1 above)
Comment (Bozydarek) [reply app/src/users/admin.py:135]: So, `Q` object should be used for `or` but not for `and`, right?

File: app/src/users/admin.py (hunk 2 above)
Comment (Bozydarek) [reply app/src/users/admin.py:145]: What would you say for renaming it to `Any`?

File: app/src/users/admin.py (hunk 1 above)
Comment (ref-humbold) [reply app/src/users/admin.py:135]: Basicly `Q` objects are used when you cannot express DB condition using keyword arguments - https://docs.djangoproject.com/en/3.2/topics/db/queries/#complex-lookups-with-q

File: app/src/users/admin.py (hunk 2 above)
Comment (ref-humbold) [reply app/src/users/admin.py:145]: `Any accommodation` sounds good. I'd then reorder lookups to: `['No accommodation', 'Any accommodation', 'Full accommodation', 'Not continuous accommodation']` (replace names with proper tuples).

File: app/src/users/admin.py [hunk 3]
```diff
@@ -116,6 +116,44 @@ def queryset(self, request, queryset):
...
 
 
+class UserAccommodationListFilter(admin.SimpleListFilter):
+    title = 'user accommodation'
+    parameter_name = 'user_accommodation'
+
+    def lookups(self, request, model_admin):
+        return [
+            ('no', 'No accommodation'),
+            ('not_continuous', "Not continuous accommodation"),
```
Comment (ref-humbold) [inline app/src/users/admin.py:126]: Please decide on quotes used ;)

Comment (Bozydarek) [reply app/src/users/admin.py:126]: I have used PaymentAcceptedListFilter as inspiration and I didn't notice it was mixed there too :upside_down_face: 
 https://github.com/ksiuwr/zosia16-site/blob/1c425dc7017ec3df320d64b5cb56a9aaa56d07e5/app/src/users/admin.py#L105-L108

File: app/src/users/admin.py (hunk 1 above)
Comment (Bozydarek) [reply app/src/users/admin.py:135]: Are you sure about that? What should be a proper syntax then?

Comment (ref-humbold) [reply app/src/users/admin.py:135]: `queryset.filter(preferences__accommodation_day_1=False, preferences__accommodation_day_2=False, preferences__accommodation_day_3=False)`
is the same as
`queryset.filter(Q(preferences__accommodation_day_1=False) & Q(preferences__accommodation_day_2=False) & Q(preferences__accommodation_day_3=False))`

**Review by ref-humbold — APPROVED** [review]

**Pull request merged**